src/include/txn.i

```diff
@@ -736,6 +736,13 @@ __wt_txn_upd_visible_type(WT_SESSION_IMPL *session, WT_UPDATE *upd)
         if (prepare_state == WT_PREPARE_LOCKED)
             continue;
 
+        if (F_ISSET(session, WT_SESSION_RESOLVING_MODIFY) && upd->txnid != WT_TXN_ABORTED &&
```

Comment:
What about consecutive modifies?

Reply:
What do you mean? It should work the same.

Reply:
Sorry, I'm probably misunderstanding. I just didn't get why we're checking specifically for a standard update here.

Reply:
We don't want to ignore history store tombstone here.

Reply:
Otherwise we still get nothing

Reply:
Oops, sorry I got confused. The modify is going to be written as a standard update in the history store.